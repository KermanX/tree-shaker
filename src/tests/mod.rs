use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
