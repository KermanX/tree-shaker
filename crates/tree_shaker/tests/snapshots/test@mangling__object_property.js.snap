---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/mangling/object_property.js
---
export function plain_object(aaa, bbb) {
	const obj = {
		a: aaa,
		b: bbb
	};
	console.log(obj.a);
	console.log(obj.b);
}
export function computed_property(unknown, aaa, bbb) {
	const key = unknown ? "a" : "b";
	const obj = {
		a: aaa,
		b: bbb
	};
	console.log(obj[key]);
}
export function property_via_destructuring(aaa, bbb) {
	const { a: foo, b: bar } = {
		a: aaa,
		b: bbb
	};
	console.log(foo);
	console.log(bar);
}
export function with_rest(__unused_7A33, bbb) {
	const { a: __unused_EBAD,...rest } = { bar: bbb };
	console.log(rest.bar);
}
export function dynamic_destructuring(unknown, aaa, bbb) {
	const { [unknown ? "a" : "b"]: value } = {
		a: aaa,
		b: bbb
	};
	console.log(value);
}
export function multi_call(aaa, bbb) {
	function f(o) {
		console.log(o.a);
	}
	f({ a: aaa });
	f({ a: bbb });
}
export function accessing_prototype(aaa) {
	return {}.toString.call(aaa);
}
export function object_assign(aaa, bbb) {
	const obj = Object.assign({}, { b: aaa }, { a: bbb });
	console.log(obj.b);
	console.log(obj.a);
}
export function dunder_proto(aaa) {
	const obj = {
		__proto__: aaa,
		prop: aaa
	};
	console.log(obj.prop);
}
